src/netlink.c

```diff
@@ -0,0 +1,431 @@
+/*
+ * This file is part of cc-oci-runtime.
+ *
+ * Copyright (C) 2016 Intel Corporation
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+/** \file
+ *
+ * Networking netlink routines, used to setup networking
+ *
+ */
+
+#include <errno.h>
+#include <string.h>
+
+#include <glib.h>
+#include <glib/gprintf.h>
+
+#include <libmnl/libmnl.h>
+#include <linux/if.h>
+#include <linux/if_link.h>
+#include <linux/rtnetlink.h>
+
+#include "netlink.h"
+#include "oci.h"
+#include "util.h"
+#include "networking.h"
+
+/*!
+ * Setup the netlink socket to use with netlink
+ * transactions. This handle should be used for all
+ * netlink transactions for a given thread.
+ *
+ * The handle is not thread safe
+ *
+ * \return \c handle to netlink on success, else \c NULL.
+ */
+struct netlink_handle *
+netlink_init(void) {
+	struct netlink_handle *hndl = g_malloc0(sizeof(*hndl));
+
+	hndl->nl = mnl_socket_open(NETLINK_ROUTE);
+	if (hndl->nl == NULL) {
+		g_debug("mnl_socket_open %s", strerror(errno));
```

Comment:
All the calls to `g_debug()` in error paths should be atleast a `g_warning()` but probably `g_critical()`.